### ISSUE TITLE
Feature suggestion: Range API

### ISSUE THREAD

**Issue by imp** [ideasman42/rangetree-rs#8]
Is it possible to add range API ?
I.e. I would like to take, query and release a range at once.

**Comment by ideasman42** [comment]
Yes, this is listed under "further work" in the readme. I didn't add this because I didn't need this for my usage.

Note that changes like this should be possible with existing structures.
